apps/dashboard/test/models/projects_test.rb

```diff
@@ -0,0 +1,86 @@
+require 'test_helper'
+
+class ProjectsTest < ActiveSupport::TestCase
+
+  test "creates project" do
+    Dir.mktmpdir do |tmp|
+      projects_path = Pathname.new(tmp)
+      OodAppkit.stubs(:dataroot).returns(projects_path)
+      attrs = { dir: 'test_project' }
+      project = Project.new(attrs)
+      project.save
+
+      assert Dir.entries("#{projects_path}/projects").include?("test_project")
+    end
+  end
+
+  test "creates .ondemand configuration directory" do
+    Dir.mktmpdir do |tmp|
+      projects_path = Pathname.new(tmp)
+      OodAppkit.stubs(:dataroot).returns(projects_path)
+      attrs = { dir: 'test_project' }
+      project = Project.new(attrs)
+      project.save
+
+      dot_ondemand_path = Pathname.new("#{projects_path}/projects/#{project.dir}")
+
+      assert Dir.entries(dot_ondemand_path).include?(".ondemand")
+    end
+  end
+
+  test "creates manifest.yml in .ondemand config directory" do
+    Dir.mktmpdir do |tmp|
+      projects_path = Pathname.new(tmp)
+      OodAppkit.stubs(:dataroot).returns(projects_path)
+      attrs = { dir: 'test"authenticity_token"=>"[FILTERED]", "project"=>{"dir"=>"project4"}, "commit"=>"Save"}
+      _project' }
+      project = Project.new(attrs)
+      project.save
+      manifest_path = Pathname.new("#{projects_path}/projects/#{project.dir}/.ondemand")
+
+      assert Dir.entries(manifest_path).include?("manifest.yml")
```

Comment:
the `attrs` I think is wrong ,so you can just inline them or define them another way.

Also the test doesn't assert what's in the manifest when it should.
```suggestion
      project = Project.new({dir: 'test_dir'})
      project.save
      assert_equal 'test_dir', project.dir
      manifest_path = Pathname.new("#{projects_path}/projects/#{project.dir}/.ondemand/manifest.yml")
      assert File.file?(manifest_path)
      
      # now assert that the manifest is what you expect
```

Reply:
Got it, yea I'm now seeing that the description and so on is edit only. I'm thinking those fields should be available to new.